core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -52,7 +55,36 @@ private[spark] class TaskDescription(
     val addedFiles: Map[String, Long],
     val addedJars: Map[String, Long],
     val properties: Properties,
-    val serializedTask: ByteBuffer) {
+    private var serializedTask_ : ByteBuffer,
+    private var task_ : Task[_] = null) extends  Logging {
```

Comment:
yeesh ok 100 is a lot! (You may have noticed I deleted one of the test suites in that PR to sidestep the compilation issues.)  @witgo why don't you implement @squito's suggestions and then we can see where things stand.  It would be helpful to also add a comment on line 59 (so it will be right above the "private var task_") saying that the variable is only used on the master.